src/resolvers/registries/npm-resolver.js

```diff
@@ -107,8 +107,19 @@ export default class NpmResolver extends RegistryResolver {
 
     const files = await this.config.getCache('cachedPackages', async (): Promise<Array<string>> => {
       const files = await fs.readdir(cacheFolder);
-      const validFiles = [];
+      const scoped_files = {};
+      for (const scope of files) {
+        // no hidden files
+        if (scope[0] === '.') {
+          continue;
+        }
+
+        if (!await fs.exists(path.join(cacheFolder, scope, METADATA_FILENAME))) {
+          scoped_files[scope] = await fs.readdir(path.join(cacheFolder, scope));
+        }
```

Comment:
Rather than checking if the file exists, you should instead check whether the folder name starts with a `@`, cf [this code](https://github.com/yarnpkg/yarn/blob/master/src/package-linker.js#L261)

Reply:
I'd also prefer if you could iterate over the scope folder inside this loop, instead of storing the result inside `scoped_files` (which should be `scopedFiles` anyway :) and iterating over it later on.